File: src/main/java/dev/alexengrig/leetcode/_560_subarray_sum_equals_k/MapSolution.java
package dev.alexengrig.leetcode._560_subarray_sum_equals_k;

import java.util.HashMap;
import java.util.Map;

class MapSolution implements Task {
    @Override
    public int subarraySum(int[] nums, int k) {
        int count = 0, sum = 0;
        Map<Integer, Integer> counts = new HashMap<>();
        counts.put(0, 1);
        for (int num : nums) {
            sum += num;
            count += counts.getOrDefault(sum - k, 0);
            counts.merge(sum, 1, Integer::sum);
        }
        return count;
    }
}

File: src/main/python/242_valid_anagram/map_solution.py
from collections import Counter

from task import Task


class MapSolution(Task):
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        counts = Counter(s)
        for ch in t:
            if ch in counts:
                if counts[ch] == 1:
                    counts.pop(ch)
                else:
                    counts[ch] -= 1
        return len(counts) == 0
